src/bot.py

```diff
@@ -257,19 +261,28 @@ def on_create_game(self):
 
     def on_start_from_town(self):
         self._curr_loc = self._town_manager.wait_for_town_spawn()
-
         # Handle picking up corpse in case of death
         if (corpse_present := is_visible(ScreenObjects.Corpse)):
             self._previous_run_failed = True
             view.pickup_corpse()
             wait_until_hidden(ScreenObjects.Corpse)
             belt.fill_up_belt_from_inventory(Config().char["num_loot_columns"])
+        if not self._hotkeys_discovered:
+            saved_games_folder = Config().general["saved_games_folder"]
+            key_file = Config().general["key_file"]
+            hotkeys.discover_hotkey_mappings(saved_games_folder, key_file)
+            self._hotkeys_discovered = True
         self._char.discover_capabilities()
-        if corpse_present and self._char.capabilities.can_teleport_with_charges and not self._char.select_tp():
-            keybind = Config().char["teleport"]
+        teleport_selected = skills.select_tp(hotkeys.right_skill_key_map[SkillName.Teleport])
```

Comment:
Why do you need to pass `hotkeys.right_skill_key_map[SkillName.Teleport]` as an argument? Can't you just use bake this hotkey into `select_tp()`?

Reply:
We could do that. I think that's beyond the scope of this, though, because that'd be refactoring the seams of the code.

That's to say that prior to my change we passed in Config().char["teleport"] and my change aims to hotswap that out with hotkeys.right_skill_key_map[SkillName.Teleport]